### PR TITLE
Adds cookbook recipe for using input filters in forms

### PR DESCRIPTION
**Pull request by froschdesign**

### CODE REVIEW

File: docs/book/cookbook/input-filter-in-forms.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+# Using Input Filters in Forms of laminas-form
+
+The following examples shows _two_ potential use cases of laminas-inputfilter within forms of [laminas-form](https://docs.laminas.dev/laminas-form/).
```
Comment (weierophinney) [inline docs/book/cookbook/input-filter-in-forms.md]:
```suggestion
The following examples shows _two_ potential use cases of laminas-inputfilter within [laminas-form](https://docs.laminas.dev/laminas-form/).
```

File: docs/book/cookbook/input-filter-in-forms.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+## Define the Input Filter in a Form
+
+An input filter can be [defined directly in a form class](https://docs.laminas.dev/laminas-form/v3/quick-start/#hinting-to-the-input-filter) itself, using `Laminas\InputFilter\InputFilterProviderInterface`.
+This interface provides one method (`getInputFilterSpecification()`) that is used by a form of laminas-form to create an input filter.
```
Comment (weierophinney) [inline docs/book/cookbook/input-filter-in-forms.md]:
```suggestion
This interface provides one method (`getInputFilterSpecification()`) that is used by a form to create an input filter.
```

Removing a redundancy; it's implied by this point due to the introduction.

File: docs/book/cookbook/input-filter-in-forms.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    }
+}
+</code></pre>
+<!-- markdownlint-enable MD033 -->
+
+## Adding an Input Filter Defined as a Separate Class to a Form
+
+### Create Input Filter
+
+[Create an input filter as separate class](../intro.md), e.g. `module/Album/src/InputFilter/AlbumInputFilter.php`:
```
Comment (weierophinney) [inline docs/book/cookbook/input-filter-in-forms.md]:
```suggestion
[Create an input filter as a separate class](../intro.md), e.g. `module/Album/src/InputFilter/AlbumInputFilter.php`:
```

File: docs/book/cookbook/input-filter-in-forms.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
+# Using Input Filters in Forms of laminas-form
+
+The following examples shows _two_ potential use cases of laminas-inputfilter within forms of [laminas-form](https://docs.laminas.dev/laminas-form/).
+
+## Define the Input Filter in a Form
+
+An input filter can be [defined directly in a form class](https://docs.laminas.dev/laminas-form/v3/quick-start/#hinting-to-the-input-filter) itself, using `Laminas\InputFilter\InputFilterProviderInterface`.
+This interface provides one method (`getInputFilterSpecification()`) that is used by a form of laminas-form to create an input filter.
+
+[Create a form as separate class](https://docs.laminas.dev/laminas-form/v3/quick-start/#factory-backed-form-extension) using the [`init` method](https://docs.laminas.dev/laminas-form/v3/advanced/#initialization), implement the interface `Laminas\InputFilter\InputFilterProviderInterface` and define the inputs via configuration array, e.g. `module/Album/src/Form/AlbumForm.php`:
```
Comment (weierophinney) [inline docs/book/cookbook/input-filter-in-forms.md]:
```suggestion
[Create a form as a separate class](https://docs.laminas.dev/laminas-form/v3/quick-start/#factory-backed-form-extension), define the [`init` method](https://docs.laminas.dev/laminas-form/v3/advanced/#initialization), implement the interface `Laminas\InputFilter\InputFilterProviderInterface`, and define its inputs via a configuration array; as an example, consider the following definition in a file found at `module/Album/src/Form/AlbumForm.php`:
```

File: docs/book/cookbook/input-filter-in-forms.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        // …
+    }
+}
+```
+
+### Create Form and Add Input Filter
+
+An input filter can be added directly in a form class itself, using the class name of the input filter or whatever name the input filter is registered under.
+
+Create a form as separate class using the `init` method and set the input filter via the `setInputFilterByName()` method of `Laminas\Form\Form`, e.g. `module/Album/src/Form/AlbumForm.php`:
```
Comment (weierophinney) [inline docs/book/cookbook/input-filter-in-forms.md]:
```suggestion
Create a form as a separate class, define its `init` method, and set the input filter via the `setInputFilterByName()` method of `Laminas\Form\Form`, e.g. `module/Album/src/Form/AlbumForm.php`:
```

**Review by weierophinney — CHANGES_REQUESTED** [review]
Minor changes, but looks good!

**Comment by froschdesign** [comment]
@weierophinney 
A final review is needed.
Thanks in advance! 👍🏻

File: docs/book/cookbook/input-filter-in-forms.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
+# Using Input Filters in Forms of laminas-form
+
+The following examples shows _two_ potential use cases of laminas-inputfilter within [laminas-form](https://docs.laminas.dev/laminas-form/).
```
Comment (weierophinney) [inline docs/book/cookbook/input-filter-in-forms.md]:
```suggestion
The following examples show _two_ potential use cases of laminas-inputfilter within [laminas-form](https://docs.laminas.dev/laminas-form/).
```

**Review by weierophinney — COMMENTED** [review]
One minor language change, but otherwise good go go!

**Pull request merged**